### ISSUE TITLE
Arguments number does not match in Hetero-gnn model

### ISSUE THREAD

**Issue by YAndrewL** [pyg-team/pytorch_geometric#3200]
Hi, I'm now trying to create a HeteroConv class using a spatial GCN.
Here's a structure of my data:
```python
Batch(
  author={
    x=[263, 566],
    pos=[263, 3],
    y=[263],
    batch=[263],
    ptr=[13]
  },
  paper={
    x=[462, 566],
    pos=[462, 3],
    batch=[462],
    ptr=[13]
  },
  (author, to, author)={ edge_index=[2, 759] },
  (paper, to, paper)={ edge_index=[2, 2818] },
  (author, to, paper)={ edge_index=[2, 707] },
  (paper, to, author)={ edge_index=[2, 707] }
)
```

And the spatial GCN is described [here](https://arxiv.org/abs/1909.05310) with a few modifications. The detailed codes are listed below, which is almost the same as the official implementation.

```python
class SpatialGraphConv(MessagePassing):
    def __init__(self, coors, in_channels, out_channels, dropout=0):
        """
        coors - dimension of positional descriptors (e.g. 2 for 2D images)
        in_channels - number of the input channels (node features)
        out_channels - number of the output channels (node features)
        hidden_size - number of the inner convolutions
        dropout - dropout rate after the layer
        """
        super(SpatialGraphConv, self).__init__(aggr='add')
        self.dropout = dropout

        self.lin_in = torch.nn.Linear(coors, in_channels)
        self.lin_out = torch.nn.Linear(in_channels, out_channels)
        self.in_channels = in_channels

    def forward(self, x: tuple, edge_index, pos):
        """
        x - feature matrix of the whole graph [num_nodes, label_dim]
        pos - node position matrix [num_nodes, coors]
        edge_index - graph connectivity [2, num_edges]
        """

        # follow the implementation of hetero-gnn
        x_src, x_dst = x
        num_nodes = min(x_src.size(0), x_dst.size(0))
        edge_index, _ = add_self_loops(edge_index, num_nodes)
        return self.propagate(edge_index=edge_index, x=x, pos=pos, aggr='add')  # [N, out_channels, label_dim]

    def message(self, pos_i, pos_j, x_j):
        """
        pos_i [num_edges, coors]
        pos_j [num_edges, coors]
        x_j [num_edges, label_dim]
        """

        relative_pos = pos_j - pos_i  # [n_edges, hidden_size * in_channels]
        spatial_scaling = F.relu(self.lin_in(relative_pos))  # [n_edges, hidden_size * in_channels]

        n_edges = spatial_scaling.size(0)
        # [n_edges, in_channels, ...] * [n_edges, in_channels, 1]
        result = spatial_scaling.reshape(n_edges, self.in_channels, -1) * x_j.unsqueeze(-1)
        return result.view(n_edges, -1)

    def update(self, aggr_out):
        """
        aggr_out [num_nodes, label_dim, out_channels]
        """
        aggr_out = self.lin_out(aggr_out)  # [num_nodes, label_dim, out_features]
        aggr_out = F.relu(aggr_out)
        aggr_out = F.dropout(aggr_out, p=self.dropout, training=self.training)

        return aggr_out
```

Besides, I create a hetero-gnn model:

```python
class HGCN(torch.nn.Module):

    def __init__(self,in_channels, out_channels, num_layers):
        super(HGCN, self).__init__()

        self.convs = torch.nn.ModuleList()

        for _ in range(num_layers):
            conv = HeteroConv(
                {
                    ('paper', 'to', 'author'): SpatialGraphConv(coors=3,
                                                                   in_channels=in_channels,
                                                                   out_channels=out_channels),

                }
            )
            self.convs.append(conv)
......
    def forward(self, x_dict, edge_index_dict, pos_dict):
        for conv in self.convs:
            x_dict = conv(x_dict, edge_index_dict, pos_dict)
            x_dict = {key: F.elu(x) for key, x in x_dict.items()}
      ......
```

And in train block, I pass x_dict, edge_index_dict, and pos_dict to the `HGCN` class, and the error happens `x_dict = conv(x_dict, edge_index_dict, pos_dict)` as `TypeError: forward() takes 3 positional arguments but 4 were given`, thus I'm figuring out what's wrong here, thanks a lot!

**Comment by rusty1s** [comment]
I see. Currently, `HeteroConv` does only support single node feature matrices. Let me see if I can fix this easily :)

**Comment by YAndrewL** [comment]
Oh, I do not quite understand what single node feature matrices refer to?

**Comment by rusty1s** [comment]
I proposed a fix in #3202. Currently, we were only adding edge-level features to each individual operator in `HeteroConv`, e.g., `conv(x_dict, edge_index_dict, edge_weight_dict)` while we ignored additional node-level features such as `pos_dict`. The PR fixes this (and includes a test re-implementing your use-case).

Note that you probably do not want to remove self-loops in your operator, as this operator will operate in bipartite graphs.
